core/services/capreg/types.go

```diff
@@ -0,0 +1,148 @@
+package capreg
+
+import (
+	"bytes"
+	"context"
+	"math/big"
+)
+
+// CapabilityID is the unique identifier of the capability in the CR.
+// It is calculated as keccak256(abi.encode(capabilityType, capabilityVersion)).
+type CapabilityID = string
+
+// DON represents a DON in the Capability Registry.
+type DON struct {
+	// ID is the unique identifier of the DON in the CR.
+	ID uint32
+
+	// IsPublic indicates whether this DON's capabilities can be accessed publicly.
+	IsPublic bool
+
+	// Nodes is the list of nodes in this DON, represented by their RageP2P public keys/IDs.
+	Nodes [][]byte
+
+	// CapabilityConfigurations are the configurations of the various capabilities of this DON.
+	CapabilityConfigurations []CapabilityConfiguration
+}
+
+// CapabilityConfiguration represents the configuration of a capability in the Capability Registry.
+// This can be considered shared configuration among all DONs that have this capability.
+type CapabilityConfiguration struct {
+	// CapabilityID is the unique identifier of the capability in the CR that this configuration is for.
+	CapabilityID          CapabilityID
+	OnchainConfigVersion  uint64
+	OnchainConfig         []byte
+	OffchainConfigVersion uint64
+	OffchainConfig        []byte
```

Comment:
@makramkd What differentiates on-chain and off-chain config? (I know this is borrowed from OCR, but I've always found it a bit confusing that both on-chain and off-chain config live on-chain -- maybe it's worth revisiting this with our generic configuration struct)?

Reply:
cc @DeividasK I believe this is covered in the Capability Registry doc

Reply:
Actually we've revisited this offline and this will most likely change.